libvncclient/rfbproto.c

```diff
@@ -1769,6 +1772,73 @@ SendExtendedKeyEvent(rfbClient* client, uint32_t keysym, uint32_t keycode, rfbBo
 }
 
 
+/*
+ * sendClientCutTextNotify
+ * it is needed when client send utf8 clipboard data
+ */
+
+static rfbBool
+sendExtClientCutTextNotify(rfbClient *client)
+{
+  rfbClientCutTextMsg cct = {0, };
+  const uint32_t be_flags = rfbClientSwap32IfLE(rfbExtendedClipboard_Notify
+                                                | rfbExtendedClipboard_Text); /*text and notify*/
+  cct.type = rfbClientCutText;
+  cct.length = rfbClientSwap32IfLE(-(sizeof(be_flags)));/*flag*/
```

Comment:
I think this needs an explaining comment regarding the minus and linking to https://github.com/rfbproto/rfbproto/blob/master/rfbproto.rst#extended-clipboard-pseudo-encoding :-)

Reply:
this behavior based on the  fact that notify is used to indicate who are provide the clipboard. So we need  first notify that we are get clipboard and then we provide the clipboard. It is tricky and not easy to  read between spec....

Reply:
yeah i meant: please add an explaining comment for future developers reading the code :-)

Reply:
I tried to add more comments....